weave/wandb_interface/wandb_lite_run.py

```diff
@@ -75,6 +77,10 @@ def __init__(
             _use_async_file_stream
             and os.getenv("WEAVE_DISABLE_ASYNC_FILE_STREAM") == None
         )
+        self._fetch_last_history_step = _fetch_last_history_step
```

Comment:
this should always be equal to `not _use_async_file_stream` since that is the only time step is used.

Reply:
`_use_async_file_stream` is set to false here by default. Which means in our commit case, it would always fetch the last history step and make the `RunHistoryKeys` query.

Reply:
this still is bleeding internal implementation details. In general, you want the parameters of something to change the effect it has, not how it works. Here, fetching history step is an internal piece of logic. I don't think the caller should be setting this flag. The implementation itself should be smart enough to only fetch the step once and only if needed to perform the operation where it is needed.

Reply:
This class is used in two different contexts: for StreamTables and to upload artifacts. We are using it for the latter. How will the class know which context it's instantiated in? In our context, we _never_ need to fetch the history keys. And we _never_ need to upsert a project. These are queries that should never happen in our context, not even once. Hence why I created underscored boolean flags as arguments. What's the proposed solution?